src/Form/Type/FormatterType.php

```diff
@@ -195,14 +195,19 @@ public function configureOptions(OptionsResolver $resolver)
 
         $formatters = [];
         foreach ($pool->getFormatters() as $code => $instance) {
-            $formatters[$code] = $translator->trans($code, [], 'SonataFormatterBundle');
+            $formatters[$translator->trans($code, [], 'SonataFormatterBundle')] = $code;
         }
 
         $formatFieldOptions = [
             'choices' => $formatters,
         ];
 
         if (count($formatters) > 1) {
+            // choice_as_value options is not needed in SF 3.0+
+            if (method_exists(FormTypeInterface::class, 'setDefaultOptions')) {
```

Comment:
Don't you need to add a use for this?

Reply:
Good catch!

Reply:
Why not use the `defined` trick here?

Reply:
Not sure if we can use this here. The `OptionsResolver` handles the options for `FormatterType`, not the options of `ChoiceType`

Reply:
ok fair enough